include/swift/Driver/DependencyGraph.h

```diff
@@ -121,6 +121,13 @@ class DependencyGraphImpl {
   llvm::StringMap<std::pair<std::vector<DependencyEntryTy>, DependencyMaskTy>> Dependencies;
 
   /// The set of marked nodes.
+  /// "Marked" means that everything provided by this node (i.e. Job) is
+  /// dirty. Thus any file using any of these provides must be recompiled.
+  /// (Only non-private entities are output as provides.)
```

Comment:
This particular note doesn't really belong here. The dependency graph (and the driver in general) doesn't know about what in a file gives rise to links in the graph, and if the logic *did* output private entities as "provides", they'd be marked differently somehow anyway.